workflow/controller/dag.go

```diff
@@ -222,10 +224,13 @@ func (d *dagContext) hasMoreRetries(node *wfv1.NodeStatus) bool {
 	if err != nil {
 		return false
 	}
-	if tmpl.RetryStrategy != nil && tmpl.RetryStrategy.Limit != nil && int32(len(node.Children)) > *tmpl.RetryStrategy.Limit {
-		return false
-	}
-	return true
+
+	nullLogger := logrus.New()
+	nullLogger.SetOutput(ioutil.Discard)
+	woc := wfOperationCtx{wf: d.wf.DeepCopy(), log: logrus.NewEntry(nullLogger)}
```

Comment:
We have to create a dummy context to be able to use `processNodeRetries`. This could be expensive since we run a `DeepCopy` on the workflow object

Reply:
Exploring the feasibility of this solution and others

Reply:
Went with an alternative solution